src/Microsoft.Health.Dicom.Core/Features/Store/StoreResponseBuilder.cs

```diff
@@ -84,9 +96,28 @@ public void AddSuccess(DicomDataset dicomDataset, ushort? warningReasonCode = nu
             { DicomTag.ReferencedSOPClassUID, dicomDataset.GetFirstValueOrDefault<string>(DicomTag.SOPClassUID) },
         };
 
-        if (warningReasonCode.HasValue)
+        if (!_enableDropInvalidDicomJsonMetadata)
         {
-            referencedSop.Add(DicomTag.WarningReason, warningReasonCode.Value);
+            if (warningReasonCode.HasValue)
+            {
+                referencedSop.Add(DicomTag.WarningReason, warningReasonCode.Value);
+            }
+        }
+        else
+        {
+            // add comment Sq / list of warnings here
+
+            var warningList = storeValidationResult.Errors.Select(
```

Comment:
nit: seems odd to have a variable with `List` in the name which is actually just an array. Could call this variable simply `warnings`.